### DISCUSSION TITLE
.NET MAUI updates in .NET 9 Preview 1

### DISCUSSION THREAD

**Discussion by davidortinau** [dotnet/maui#20558] [Announcements]
The team is continuing work on core fundamentals of the .NET MAUI SDK to improve overall product quality. This includes expanding test coverage, end to end scenario testing, and bug fixing. 

.NET MAUI updates in .NET 9 Preview 1:
* [Release notes](https://github.com/dotnet/core/blob/main/release-notes/9.0/preview/preview1/dotnetmaui.md) 
* [GitHub Release](https://github.com/dotnet/maui/releases/tag/9.0.100-preview.1.9973)
* [How to install](https://github.com/dotnet/core/blob/main/release-notes/9.0/install-workloads.md)

.NET 9 Preview 1:
* [Discussion](https://aka.ms/dotnet/9/preview1)
* [Release notes](https://github.com/dotnet/core/blob/main/release-notes/9.0/preview/preview1/README.md) 

> **Known Issue:** building `net8.0-*` target frameworks from .NET 9 does not work in preview 1.

**Comment by bcaceiro** [comment]
Can you post the information on how to start using this preview?

**Reply by davidortinau** [reply]
https://github.com/dotnet/core/blob/main/release-notes/9.0/install-workloads.md

Install .NET 9 from the installer, and then `dotnet workload install maui`.

**Comment by nor0x** [comment]
thanks @davidortinau 
what is the vision and roadmap for the .NET9 timeline? Any news about Swift Interop or Platform Channels?

**Reply by davidortinau** [reply]
Roadmap and link to the Swift Interop design. https://github.com/dotnet/maui/wiki/Roadmap

These aren't on the roadmap, so you can find more info here:
- Platform channels https://github.com/Redth/Microsoft.Maui.Platform.Channels
- Slim bindings https://github.com/Redth/MapboxSlimBindingDemo

We have a couple new examples of slim bindings coming soon.

**Comment by MSU-PV** [comment]
Hi guys, we are ready for **MAUI** in **VisionOS**, how about you?

![20240305_143847](https://github.com/dotnet/maui/assets/119041038/78cffa88-373e-4e8b-be76-627b71098da6)

Do you have any good news for me? 😊

**Reply by jonmdev** [reply]
Wow you have a neater desk than I do. My desk looks post apocalyptic. 😂

**Reply by MSU-PV** [reply]
😂

**Comment by jonmdev** [comment]
@davidortinau Any chance we can get:

- [Worker Editor on iOS](https://github.com/dotnet/maui/issues/17757)? 
- [Working Label on Android](https://github.com/dotnet/maui/issues/17884)? 
- [Working Label Measurement on Android](https://github.com/dotnet/maui/issues/18328) (likely cause of above issue)?
- [Working Border on Android](https://github.com/dotnet/maui/issues/18071)?

I don't mean to be a nag about this. I appreciate your team's work, but I don't know how anyone can release any apps with these problems. I am certainly still stuck.

As far as I can tell those are the only remaining crippling issues in the most basic types of elements. If I could get those working and the current ongoing Shadow fixes I see end up working as advertised, I would have everything I need to function.

Thanks for any thoughts or timeline.
